src/duke/Deadline.java

```diff
@@ -0,0 +1,71 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * This is a child class of duke.Task, duke.Deadline.
+ * duke.Deadline accepts another variable, 'by' that
+ * stores the time this duke.Deadline has to be completed
+ *
+ * @author  Hsiao Jiet
+ * @version 1.0
+ * @since   2022-1-15
+ */
+
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Creates duke.Deadline object with provided LocalDate variable
+     */
+    public Deadline(String name, int isDone, LocalDate by) {
+        super(name, isDone);
+        super.type = 'D';
+        this.by = by;
+    }
+
+    /**
+     * Creates duke.Deadline object without provided LocalDate variable
+     * Used when there is the need to parse text into date first
+     */
+    public Deadline(String name, int isDone) {
+        super(name, isDone);
+        super.type = 'D';
+    }
+
+    public void setBy(LocalDate by) {
+        this.by = by;
+    }
+
+    /**
+     * Setting duke.Deadline object's date it should end by
+     */
+    public void setStringToLocalDate(String date) {
```

Comment:
What's your rationale for naming this method as `setString...`? Do you think it is hard to tell what "String" the caller of the function is setting?